src/components/Picture.astro

```diff
@@ -0,0 +1,30 @@
+---
+type Props = {
+  className?: string
+  src: string
+  alt: string
+}
+
+const medias = [
+  {
+    '(min-width: 1024px)': 'desktop'
+  },
+  {
+    '(min-width: 768px)': 'tablet'
+  }
+]
+
+const { className, src, alt } = Astro.props
+---
+
+<picture>
```

Comment:
Please allow us to send ID and Class (props) if we need to the Picture Element. I see we have className. I would use that one here, and create a new one for the img tag itself